### PR TITLE
Upgrade improvements

### PR DESCRIPTION
**Pull request by Anakin5**
- Add upgrade_level parameter
- Fix a mistake in upgrade cost
- Some refactoring

### CODE REVIEW

File: pokemongo_bot/cell_workers/pokemon_optimizer.py [hunk 1]
```diff
@@ -154,14 +155,15 @@ def work(self):
             xp_all += xp
 
         self.apply_optimization(transfer_all, evolve_all, upgrade_all, xp_all)
-        inventory.update_web_inventory()
```
Comment (solderzzc) [inline pokemongo_bot/cell_workers/pokemon_optimizer.py]: Is this deleted by merge issue?

Comment (Anakin5) [inline pokemongo_bot/cell_workers/pokemon_optimizer.py]: No it is just not needed. Web inventory is saved at next loop.

**Comment by solderzzc** [comment]
👍

**Comment by z-jun** [comment]
Hi, very useful should be extension of evolve_for_xp parametr with an white/black list of pokemons/groups. Idea is to save candies for upgrades  while evolving for xp on trash mons. For example:

evolve_for_xp: [ "all", "!Eevee", "!Gym"],

**Pull request merged**